src/library/coverartcache.cpp

```diff
@@ -7,9 +7,42 @@
 #include "library/coverartcache.h"
 #include "library/coverartutils.h"
 
-// Large cover art wastes space in our cache when we typicaly won't show them at
-// their full size. If no width is specified, this is the maximum width cap.
-const int kMaxCoverWidth = 300;
+
+namespace {
+    QString pixmapCacheKey(quint16 hash, int width) {
+        if (width == 0) {
+            return QString("CoverArtCache_%1").arg(QString::number(hash));
+        } else {
+            return QString("CoverArtCache_%1_%2")
+                    .arg(QString::number(hash)).arg(width);
+        }
+    }
+
+    // Large cover art wastes space in our cache when we typically won't show them at
+    // their full size. If no width is specified, this is the maximum width cap.
+    const int kMaxCoverSize = 300;
+
+    // The transformation mode when scaling images
+    const Qt::TransformationMode kTransformationMode = Qt::SmoothTransformation;
+
+    // Resizes the image (preserving aspect ratio) if it is larger than
+    // maxEdgeSize on either side.
+    inline QImage limitImageSize(const QImage& image, int maxSize) {
```

Comment:
No, this is not a new function. I just renamed 2 functions as mentioned in 
the description.

Other changes:
* Defined the constant kTransformationMode

The actual implementation of the functions should not have changed. I just 
wanted to get rid of all the dependencies pulled in by the header.

If you feel more confident without any content changes at all, I can revert 
this PR and redo the transformation by only shifting the code around.

On 07/31/2015 12:12 AM, Daniel Schürmann wrote:
>
> In src/library/coverartcache.cpp 
> <https://github.com/mixxxdj/mixxx/pull/662#discussion_r35927344>:
>
> > +        } else {
> > +            return QString("CoverArtCache_%1_%2")
> > +                    .arg(QString::number(hash)).arg(width);
> > +        }
> > +    }
> > +
> > +    // Large cover art wastes space in our cache when we typically won't show them at
> > +    // their full size. If no width is specified, this is the maximum width cap.
> > +    const int kMaxCoverSize = 300;
> > +
> > +    // The transformation mode when scaling images
> > +    const Qt::TransformationMode kTransformationMode = Qt::SmoothTransformation;
> > +
> > +    // Resizes the image (preserving aspect ratio) if it is larger than
> > +    // maxEdgeSize on either side.
> > +    inline QImage limitImageSize(const QImage& image, int maxSize) {
>
> This is a new function ...
> It looks like we need a full review :-/
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/662/files#r35927344>.
>